### PR TITLE
CONTRIBUTING: notes abount adding doc for new instrumentation

### PR DESCRIPTION
**Pull request by xrmx**
# Description

While at it revise the versioning paragraph to take into account the openai instrumentation.

Refs #3060 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [ ] Followed the style guidelines of this project
- [ ] Changelogs have been updated
- [ ] Unit tests have been added
- [x] Documentation has been updated

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -272,7 +272,9 @@ Below is a checklist of things to be mindful of when implementing a new instrume
...
   - ex. <https://github.com/open-telemetry/opentelemetry-python-contrib/blob/60fb936b7e5371b3e5587074906c49fb873cbd76/instrumentation/opentelemetry-instrumentation-grpc/tests/test_aio_server_interceptor.py#L84>
-- All instrumentations have the same version. If you are going to develop a new instrumentation it would probably have `X.Y.dev` version and depends on `opentelemetry-instrumentation` and `opentelemetry-semantic-conventions` for the same version. That means that if you want to install your instrumentation you need to install its dependencies from this repo and the core repo also from git.
+- Most of the instrumentations have the same version. If you are going to develop a new instrumentation it would probably have `X.Y.dev` version and depends on `opentelemetry-instrumentation` and `opentelemetry-semantic-conventions` for a [compatible version](https://peps.python.org/pep-0440/#compatible-release). That means that you may need to install the instrumentation dependencies from this repo and the core repo from git.
+- Documentation
+  - When adding a new instrumentation remember to add an entry in `docs/instrumentation/<instrumentation>/<instrumentation>.rst`
```
Comment (xrmx) [inline CONTRIBUTING.md:277]: Here I can probably be more specific than with `an entry`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -272,7 +272,9 @@ Below is a checklist of things to be mindful of when implementing a new instrume
...
-- All instrumentations have the same version. If you are going to develop a new instrumentation it would probably have `X.Y.dev` version and depends on `opentelemetry-instrumentation` and `opentelemetry-semantic-conventions` for the same version. That means that if you want to install your instrumentation you need to install its dependencies from this repo and the core repo also from git.
+- Most of the instrumentations have the same version. If you are going to develop a new instrumentation it would probably have `X.Y.dev` version and depends on `opentelemetry-instrumentation` and `opentelemetry-semantic-conventions` for a [compatible version](https://peps.python.org/pep-0440/#compatible-release). That means that you may need to install the instrumentation dependencies from this repo and the core repo from git.
+- Documentation
+  - When adding a new instrumentation remember to add an entry in `docs/instrumentation/` named `<instrumentation>/<instrumentation>.rst` to have the instrumentation documentation referenced from the index.
```
Comment (emdneto) [inline CONTRIBUTING.md:277]: Maybe we can leave an entry template under `_template` dir, wdyt?
```suggestion
  - When adding a new instrumentation remember to add an entry in `docs/instrumentation/` named `<instrumentation>/<instrumentation>.rst` to have the instrumentation documentation referenced from the index. You can use the entry template available [here](./_template/autodoc_entry.rst)
```

Comment (emdneto) [reply CONTRIBUTING.md:277]: The template would be something like this:

```rst
.. include:: ../../../instrumentation/opentelemetry-instrumentation-<REPLACE ME>/README.rst
    :end-before: References

.. automodule:: opentelemetry.instrumentation.<REPLACE ME>
    :members:
    :undoc-members:
    :show-inheritance:
```

**Review by lzchen — APPROVED** [review]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -272,7 +272,9 @@ Below is a checklist of things to be mindful of when implementing a new instrume
...
   - ex. <https://github.com/open-telemetry/opentelemetry-python-contrib/blob/60fb936b7e5371b3e5587074906c49fb873cbd76/instrumentation/opentelemetry-instrumentation-grpc/tests/test_aio_server_interceptor.py#L84>
-- All instrumentations have the same version. If you are going to develop a new instrumentation it would probably have `X.Y.dev` version and depends on `opentelemetry-instrumentation` and `opentelemetry-semantic-conventions` for the same version. That means that if you want to install your instrumentation you need to install its dependencies from this repo and the core repo also from git.
+- Most of the instrumentations have the same version. If you are going to develop a new instrumentation it would probably have `X.Y.dev` version and depends on `opentelemetry-instrumentation` and `opentelemetry-semantic-conventions` for a [compatible version](https://peps.python.org/pep-0440/#compatible-release). That means that you may need to install the instrumentation dependencies from this repo and the core repo from git.
+- Documentation
```
Comment (aabmass) [inline CONTRIBUTING.md:276]: Please push back if this is scope creeping your PR, but should we also mention adding to `tox.ini` for the new package?

**Review by aabmass — APPROVED** [review]

**Review by emdneto — APPROVED** [review]

File: CONTRIBUTING.md (hunk 3 above)
Comment (xrmx) [reply CONTRIBUTING.md:276]: Added

**Comment by xrmx** [comment]
Update PR with reference to tox.ini and the autodoc entry to the template
